### PR TITLE
[FAN-181] Optimiz detail

### PR DESCRIPTION
**Pull request by XuMinhui**
complete pool bids and pool owner
https://fangible.atlassian.net/browse/FAN-181

### CODE REVIEW

**Comment by BigBadAlien** [comment]
Kill package-lock plz and revert yarn.lock

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 1]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                    <p>
+                      <span>{t('details-nft.token-info.name-tags')}:</span>
+                      {` ${itemName} ${itemsymbol && `(${itemsymbol})`}`}
+                    </p>
+                  )}
+                  {
+                    <p>
+                      <span>{t('details-nft.token-info.standard')}:</span>
+                      {` ${
+                        standard === NftType.ERC1155 ? 'ERC-1155' : 'ERC-721'
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: t

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 2]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                  )}
+                  {itemName && (
+                    <p>
+                      <span>{t('details-nft.token-info.name-tags')}:</span>
+                      {` ${itemName} ${itemsymbol && `(${itemsymbol})`}`}
+                    </p>
+                  )}
+                  {
+                    <p>
+                      <span>{t('details-nft.token-info.standard')}:</span>
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: move ':' to t

File: src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  title,
+  desc,
+  isScan,
+  contract,
+}: InfoTabsIsTokenInfoprops) => {
+  const classes = useInfoTabsIsTokenInfoStyled();
+
+  return (
+    <div>
+      {title && <h5 className={classes.title}>{title}</h5>}
```
Comment (BigBadAlien) [inline src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx]: Typography

File: src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      {desc && <div className={classes.desc}>{desc}</div>}
+
+      {isScan && (
+        <div
+          className={classes.button}
+          onClick={() => {
+            const href = `https://bscscan.com/address/${contract}`;
+            window.open(href);
```
Comment (BigBadAlien) [inline src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx]: Convert to anchor

File: src/modules/buyNFT/components/InfoTabsItem/useInfoTabsItemStyles.ts [hunk 5]
```diff
@@ -66,3 +66,42 @@ export const useInfoTabsItemStyles = makeStyles<Theme>(theme => ({
...
+    alignItems: 'center',
+    backgroundColor: '#fff',
+    cursor: 'pointer',
+    userSelect: 'none',
+    marginTop: 10,
+  },
+
+  btnSpan: {
+    marginLeft: 10.5,
+    fontFamily: 'Helvetica Neue',
```
Comment (BigBadAlien) [inline src/modules/buyNFT/components/InfoTabsItem/useInfoTabsItemStyles.ts]: Plz, check is the fontFamily really required here

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 6]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
             const renderedHistoryList = (
               <InfoTabsList>
-                <InfoTabsItem
-                  title="Offered 3 BNB for 1 edition"
-                  author="yeah66"
-                  date={new Date()}
-                />
+                {poolHistory.map(item => {
+                  let titleStr = '';
+                  switch (item.event) {
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: Extract function

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 7]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+
+                  return (
+                    <InfoTabsItem
+                      key={item.time}
+                      title={titleStr}
+                      author={
+                        item.sender.username ||
+                        truncateWalletAddr(item.sender.address)
+                      }
+                      date={new Date(item.time)}
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: Map it in action

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 8]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                {poolBids.map(item => {
+                  return (
+                    <InfoTabsItem
+                      key={item.time}
+                      title="Bid placed"
+                      author={
+                        item.sender.username ||
+                        truncateWalletAddr(item.sender.address)
+                      }
+                      date={new Date(item.time)}
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: map it in action

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 9]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
-                  price={new BigNumber('10')}
-                  cryptoCurrency="ETH"
-                  cryptoPrice={new BigNumber(10.55413)}
-                  href="//google.com"
-                />
+                {poolBids.map(item => {
+                  return (
+                    <InfoTabsItem
+                      key={item.time}
+                      title="Bid placed"
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: t

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 10]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                      author={
+                        item.sender.username ||
+                        truncateWalletAddr(item.sender.address)
+                      }
+                      date={new Date(item.time)}
+                      price={new BigNumber(item.price).multipliedBy('368')}
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: @XuMinhui Why it's exactly 368?

Comment (BigBadAlien) [reply src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: Item.price must be a BigNumber and mapped in action

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 11]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                    item.owner.username ||
+                    truncateWalletAddr(item.owner.address);
+
+                  return (
+                    <ProfileInfo
+                      key={item.owner.address}
+                      isTitleFirst
+                      avatarSize="big"
+                      title={showUser}
+                      subTitle={`${item.balance} copies`}
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: t

File: src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx [hunk 12]
```diff
@@ -185,93 +199,231 @@ export const BuyNFT = () => {
...
+                      users={[
+                        {
+                          name: showUser,
+                          avatar:
+                            item.owner.avatar ||
+                            'https://picsum.photos/44?random=1',
```
Comment (BigBadAlien) [inline src/modules/buyNFT/screens/BuyNFT/BuyNFT.tsx]: @Antroll Plz check the Bombist

File: src/modules/overview/actions/fetchPoolHistory.ts [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+              event: item.auction_event,
+              sender: {
+                username: item.username,
+                address: item.user_address,
+                avatar: item.avatar,
+              },
+              quantity: item.quantity,
+              price: new BigNumber(Web3.utils.fromWei(item.price)).toString(),
+              symbol: 'BNB',
+              time: item.ctime * 1000,
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolHistory.ts]: Date

File: src/modules/overview/actions/fetchPoolBids.ts [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+        return data
+          .map(item => {
+            return {
+              sender: {
+                address: item.sender,
+                avatar: item.avatar,
+                username: item.username,
+              },
+              txId: item.txid,
+              time: item.ctime * 1000,
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolBids.ts]: Date

File: src/modules/overview/actions/fetchPoolBids.ts [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+              price: new BigNumber(Web3.utils.fromWei(item.amount1)).toString(),
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolBids.ts]: Remove `toString`

File: src/modules/overview/actions/fetchPoolBids.ts [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+              symbol: 'BNB',
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolBids.ts]: Select actual token

File: src/modules/overview/actions/fetchPoolHistory.ts [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+  user_address: string;
+  username: string;
+}
+
+interface IFetchPoolHistoryParams {
+  poolId: number;
+  poolType: AuctionType;
+}
+
+interface wrapperPoolHistory {
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolHistory.ts]: IWrapperPoolHistory

File: src/modules/overview/actions/fetchPoolHistory.ts [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+  event: AUCTION_EVENT;
+  sender: IRoleInfo;
+  quantity: number;
+  price: string;
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolHistory.ts]: BigNumber

File: src/modules/overview/actions/fetchPoolHistory.ts [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+  symbol: string;
+  time: number;
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolHistory.ts]: Date

File: src/modules/overview/actions/fetchPoolNftOwner.ts [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+        console.log(data);
+        return data?.owners
+          .map(item => {
+            return {
+              owner: {
+                address: item.owneraddress,
+                avatar: item.avatar,
+                username: item.username,
+              },
+              balance: +item.balance,
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolNftOwner.ts]: BigNumber

File: src/modules/overview/actions/fetchPoolNftOwner.ts [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+        poolId: params.poolId,
+        poolType: parseInt(poolTypeMap[params.poolType]),
+      },
+    },
+    meta: {
+      auth: true,
+      driver: 'axiosSmartchain',
+      asMutation: false,
+      getData: ({ data }) => {
+        console.log(data);
```
Comment (BigBadAlien) [inline src/modules/overview/actions/fetchPoolNftOwner.ts]: Remove

File: src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+          }}
+        >
+          <BscScanIcon />
+          <span className={classes.btnSpan}>View on BscScan</span>
```
Comment (BigBadAlien) [inline src/modules/buyNFT/components/InfoTabsItem/InfoTabsIsTokenInfo.tsx]: t

**Review by Antroll — APPROVED** [review]

**Comment by Antroll** [comment]
@XuMinhui you can merge PR

**Pull request merged**